### ISSUE TITLE
Longest Peak

### ISSUE THREAD

**Issue by drkennetz** [codeconnector/CodingDojo#258]
# Longest Peak

Write a function that takes in an array of integers and returns the length of the longest peak in the array.

A peak is defined as adjacent integers in the array that are *strictly* increasing until they reach a tip (the highest value in the peak), at which point they become *strictly* decreasing. At least three integers are required to form a peak. 

I will make this abundantly clear in the examples.

## Business Rules/Errata

1. A peak *must* increase and then decrease. 
2. The input can be an empty array, and must be an array of integers.
2. If equal value numbers appear in the run of a peak, the peak is stopped and restarted.
3. A peak must contain at least 3 numbers, such as 1,2,1
4. Numbers can be positive or negative.
5. *Bonus:* Optimal time and space: O(n) time | O(1) space - where n is the length of the input array.

## Examples
```
longest_peak([5, 4, 3, 2, 1]) -> 0 # all decreasing
longest_peak([1, 2, 3, 4, 5]) -> 0 # all increasing
longest_peak([5, 4, 3, 2, 1, 2, 10, 12]) -> 0 # all decreasing, then all increasing
longest_peak([]) -> 0 # empty input check mandatory :D
longest_peak([1, 3, 2]) -> 3 # normal peak
longest_peak([1, 2, 3, 4, 5, 1]) -> 6 # peak ends on last number
longest_peak([2,2,3,2]) -> 3 # repeat at the beginning
longest_peak([1, 2, 3, 2, 1, 1]) -> 5 # repeat at the end
longest_peak([1, 2, 3, 3, 2, 1]) -> 0 # equality at peak tip, 0 is returned
longest_peak([1, 1, 1, 2, 3, 10, 12, -3, 2, 3, 45, 800, 99, 98, 0, -1, 2, 3, 4, 5, 0, -1]) -> 9 # peak in the middle -3, 2, 3, 45, 800, 99, 98, 0, -1
```

**Comment by mrumiker** [comment]
Cool problem. I noticed that the examples never tested a scenario where a peak is ended by an immediate uphill (you always had repeated numbers doing this job). Edited some of the plateaus out of the last example to remedy this.

**Comment by xanderyzwich** [comment]
It seems to be inferred but not directly said that the decline of the peak must be the same length as the incline. Is that correct?

**Comment by mrumiker** [comment]
@xanderyzwich Doesn't have to be - check out the [2, 2, 3, 2] example

**Comment by xanderyzwich** [comment]
@mrumiker That's a run of 3 [2, 3, 2] with rise and fall both being 1

**Comment by mrumiker** [comment]
@xanderyzwich You're right! From the instructions, I interpret a peak as any increasing run of numbers followed immediately by a decreasing run. Any "plateau" (a pair of equal numbers) would end the peak, as would any increasing pair occurring on the downhill portion.

**Comment by xanderyzwich** [comment]
So it seems like I'm wrong if this one is right
> longest_peak([1, 2, 3, 4, 5, 1]) -> 6 # peak ends on last number

**Comment by xanderyzwich** [comment]
closed by #272

**Comment by mrumiker** [comment]
Sorry about the mixup on this - I realize my comment where I said "You're right!" wasn't clear. I was trying to say "You're right about the example I mentioned, but I still don't think the increase and decrease have to be equal based on what the directions say." I wish I'd actually noticed the example that proved my point!
